Mathlib/Topology/Algebra/Valued/WithVal.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2025 Salvatore Mercuri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Salvatore Mercuri
+-/
+import Mathlib.Topology.UniformSpace.Completion
+import Mathlib.Topology.Algebra.Valued.ValuationTopology
+
+/-!
+# Ring topologised by a valuation
+
+`WithVal v` is a type synonym for a ring `R` equipped with the topology coming from
+a valuation.
+
+## Main definitions
+ - `WithVal` : type synonym for a ring equipped with the topology coming from a valuation.
+ - `WithVal.equiv` : The canonical ring equivalence between `WithValuation v` and `R`.
+ - `Valuation.Completion` : the uniform space completion of a field `K` according to the
+  uniform structure defined by the specified valuation.
+-/
+
+noncomputable section
+
+variable {R : Type*} {Γ₀ : outParam (Type*)} [Ring R] [LinearOrderedCommGroupWithZero Γ₀]
```

Comment:
I don't think the codomain can be an outParam since one might have several different G0s for the same R. outParam only works when R completely determines G0.